[Search/RoadMap] Some improvements & bug fix

indydedeken opened the pull request:
- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
  setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.
- Geolocation is now waiting for stationStorage is loaded in order to
  setPositionMarker on the roadMap.

indydedeken committed:
[Search view] Reorganization of the code

Geolocation is now waiting for stationStorage is loaded in order to
setPositionMarker on the roadMap.

indydedeken committed:
[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

- Del useless function call

Del useless function call

indydedeken committed:
[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

indydedeken committed:
[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

indydedeken committed:
[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

eliemichel commented:
Why do you need `stationStorage` to add a `positionMarker` on the `roadMap`?

eliemichel commented on js/Control.OSMGeocoder.js:
```diff
@@ -19,7 +19,7 @@ L.Control.OSMGeocoder = L.Control.extend({
 
 			// IDE - Map have the power to draw a marker
-			RoadMap.addMarkerSearch(this._map.fitBounds(bounds));
+			RoadMap.setRequestedPositionMarker(this._map.fitBounds(bounds));
```
You should not modify vendor libraries as they could be updated.
Actually, as soon as you modify a library, you have to care about its updates to merge your modification at any time, i.e. you have to maintain it. So we better avoid it!

indydedeken replied:
Damage is already done. Here it's a simple function renaming.

Can I open an issue an fix it later ? Vendor code is not so much impacted.

Phyks replied:
If you simply want to rename the function, you can also define an alias in the code :)

As I think this vendor code was already modified in the past, I think it is reasonable to open an issue and revert all the vendor code later.

eliemichel replied:
Yep, I will open a new issue.

indydedeken commented:
I missed it, bad copy-paste. I will delete it.

indydedeken committed:
Del useless function call

indydedeken committed:
Merge branch 'design' of https://github.com/indydedeken/VelibFxos into design

indydedeken committed:
Merge branch 'design' of https://github.com/indydedeken/VelibFxos into design

indydedeken committed:
Merge branch 'design' of https://github.com/indydedeken/VelibFxos into design

indydedeken committed:
Merge remote-tracking branch 'origin' into design

indydedeken committed:
[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

- Del useless function call

Del useless function call

[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

indydedeken committed:
[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

- Del useless function call

Del useless function call

[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

[RoadMap] update function setRequestedPositionMarker

- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.

Del useless function call

indydedeken committed:
Revert "[RoadMap] update function setRequestedPositionMarker"

This reverts commit 0b05774677164931b73035e666a0f063853af5bf.

indydedeken committed:
x

indydedeken committed:
Revert "[RoadMap] update function setRequestedPositionMarker"

This reverts commit 9a07ccbe9944298e24ffe191a5c7a7560675d8b4.

eliemichel commented on js/RoadMap.js:
```diff
@@ -62,47 +63,53 @@ var RoadMap = (function() {
 
-        for(var i=0; i < stations.length; i++) {
+        // build Markers
+        for(var i=0; i < stationStorage.length; i++) {
```
Does `stationStorage` really have a `length` attribute?

eliemichel replied:
Oh, there is a `stationStorage = stationStorage.getStations()`.
Please avoid reusing the same variable name when data type changes.
In the right-hand member, `stationStorage` is a `StationStorage` (this is why it is called like so). But in the left-hand member, it is a list of stations.

It is really confusing actually, and `stationStorage` is not a good name for a list of stations, as it is already associated with another type of object. You could name it `stations` or `stationList` for instance.

Furthermore, it breaks a global variable (within the Black Box scope) which could be used in some other places as the original instance of `StationStorage`. This is an unexpected side effect.
